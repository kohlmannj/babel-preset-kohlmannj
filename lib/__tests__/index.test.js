const { transformSync } = require('@babel/core');
const presetKohlmannj = require('..');

describe('babel-preset-kohlmannj', () => {
  it('should load default presets and plugins', () => {
    expect(presetKohlmannj()).toMatchSnapshot();
  });

  it('loads @babel/preset-typescript', () => {
    expect(presetKohlmannj({ env: () => 'development' }, { typescript: true })).toMatchSnapshot();
  });

  it('transforms React JSX code when passing react: true', () => {
    const source = `<div>Hello World</div>`;

    expect(() => {
      transformSync(source, {
        babelrc: false,
        presets: [[presetKohlmannj, { react: true }]],
      });
    }).not.toThrowError();
  });

  it('transforms TypeScript code when passing typescript: true', () => {
    const source = `interface IDingus { foo: boolean; bar: number; }`;

    expect(() => {
      transformSync(source, {
        babelrc: false,
        presets: [[presetKohlmannj, { typescript: true }]],
      });
    }).not.toThrowError();
  });
});
