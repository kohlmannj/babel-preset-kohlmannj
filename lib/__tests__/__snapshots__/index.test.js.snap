// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-preset-kohlmannj loads @babel/preset-typescript 1`] = `
Object {
  "plugins": Array [
    "@babel/plugin-proposal-class-properties",
    "@babel/plugin-proposal-object-rest-spread",
    Array [
      "@babel/plugin-proposal-decorators",
      Object {
        "legacy": true,
      },
    ],
    "@babel/plugin-proposal-export-namespace-from",
    "@babel/plugin-proposal-function-sent",
    "@babel/plugin-proposal-json-strings",
    "@babel/plugin-proposal-numeric-separator",
    "@babel/plugin-proposal-throw-expressions",
    "@babel/plugin-syntax-dynamic-import",
    "@babel/plugin-syntax-import-meta",
  ],
  "presets": Array [
    Array [
      "@babel/preset-env",
      Object {
        "debug": false,
        "modules": false,
        "useBuiltIns": "entry",
      },
    ],
    "@babel/preset-typescript",
  ],
}
`;
